src/tools/roc_recv/main.cpp

```diff
@@ -41,6 +43,15 @@ int main(int argc, char** argv) {
 
     pipeline::ReceiverConfig config;
 
+    if (args.list_drivers_given) {
```

Comment:
I'd move this `if` before `pipeline::ReceiverConfig config;` and return from main() at the end:

```
    if (args.list_drivers_given) {
        ...
        return 0;
    }
```

Reply:
The same for roc_send version.

Reply:
Just curious, why have this before the ReceiverConfig config?

Reply:
Just to group the code logically and move the config declaration closer to the code that is filling it. So it will look like:

```
if (args.list_drivers_given) {
    // the code unrelated to config
    return 0;
}

pipeline::ReceiverConfig config;

// the code filling the config
```

A general (quite common) recommendation is to declare variables as close as possible to their first use. This is a special case of an even more general recommendation to declare variables as locally as possible.